### DISCUSSION TITLE
CRCs, SHA256 and GUIDs

### DISCUSSION THREAD

**Discussion by 737simpilot** [CIDRAM/CIDRAM#295] [Q&A] [answered]
I've been messing around with creating GUIDs for things in Cyberchef. If you're using CRC, do you know what the pros and cons would be between using CRC and GUID instead? Is it more on the lines of unique integrity values or something? Also, curios why SHA256 wouldn't be used? I'm thinking too CPU heavy? I mention SHA256 because it's used in cryptocurrency, encryption and my favorite website VirusTotal which is my main means of anti-virus. I don't run over bloated, TLS MiTM spying nonsense traditional anti-virus nonsense anymore. LOL I check **ALL** file downloads with their hashes or file uploads at VirusTotal. I use HashTools and it has a VirusTotal check ability. I actually edited the VirusTotal Wikipedia page so check that out. Great thing about VirusTotal is the Relations and Behaviors pages. Very interesting Info. on files. And it's great source for IP information for research.

**Comment by Maikuolan** [comment, marked as answer]
> If you're using CRC, do you know what the pros and cons would be between using CRC and GUID instead?

The two main drawbacks to CRC that immediately come to mind are:
1. It can be a little *too* fast sometimes, depending on the context. Not enough CPU cycles required to produce a CRC means that cracking/reversing a CRC, or finding collisions for a CRC, are a little too easy to make it viable for cryptographic purposes (which is why nobody uses it for passwords or for encryption). However, in that regard, it might still be okay for basic file checksums and integrity purposes, seeing as being "too fast" isn't normally a problem in that context (and sometimes, in fact, you may want something a little faster).
2. Not all CRC implementations utilise the same mathematical process, therefore meaning that not all CRC implementations necessarily always produce the same hashes for the same data. If you're always using the same implementation, using CRCs only internally, not dealing with external CRCs from other implementations and so on, then that doesn't really matter anyway, because all your hashes will still be from the "same implementation" (thus always *internally* consistent, at least). However, if you're using different implementations in conjunction (e.g., hashes produced by PHP alongside hashes produced by WinRAR/7-zip/WinZip/etc), or hashes from external sources that might've been produced using different or unknown implementations, there'll always exist some small degree of risk that you'll end up with hashes that don't match, even if based upon the same source data. In theory, this problem (i.e., using different mathematical processes, thus inconsistent hashes) could exist for implementations of *any* hash algorithm, so it's not so much a matter based in theory, but rather that I've actually encountered that not all implementations are the same across the board for this particular algorithm in a number of popular, well-known software packages.

> Also, curios why SHA256 wouldn't be used? I'm thinking too CPU heavy?

As for why Cyberchef doesn't use SHA256, I have no idea. I'm not too familiar with Cyberchef, so I couldn't speculate on their reasons for/against a particular algo like that.

CIDRAM prefers SHA256 for it's hashes though. :-)

> I don't run over bloated, TLS MiTM spying nonsense traditional anti-virus nonsense anymore. LOL I check **ALL** file downloads with their hashes or file uploads at VirusTotal.

Exactly the same for me, too. And, I've found most anti-virus packages available nowadays to be more trouble than they're worth, causing all kinds of problems for my hardware and my machines, more often than not failing to actually catch anything anyway.

**Comment by 737simpilot** [comment]
Very interesting about CRC. Thanks for sharing your knowledge on that. I could have sworn CIDRAM used CRC, but wasn't too sure. And now that I think of it, it is SHA256 as you mention. _slaps forehead_ LOL

No, I believe Cyberchef  has SHA256 capability. It's a pretty cool bit of code there.  Cyberchef is hosted here at Github and rendered browser side and you can go CLI I guess in Linux and maybe Windows. So the capabilities are pretty interesting. I just don't know how to use the CLI in Linux to do some of the things I'd like to do with Cyberchef.


>Exactly the same for me, too. And, I've found most anti-virus packages available nowadays to be more trouble than they're worth, causing all kinds of problems for my hardware and my machines, more often than not failing to actually catch anything anyway.

You're the one and only person that I know of that has said they do the same thing I do. I've been telling lots of people about my "unorthodox" approach to malware mitigation with the use of Sandboxie, uBlock, and VirusTotal, but it either falls on deaf years or they don't understand it all. If I knew C or some variant of C like C#, I had the idea of coding  an open source anti-virus using VirusTotal. But being the person I am with a lot of foresight I see some eventual potential issues on the horizon. Especially in terms of a monetary aspect with the anti-virus engines, API and what not. I do have programs _like_ this that use VirusTotal, but it's not exactly what I have in mind. 

Anyway, thanks for that insightful bit of information there. And sorry about the confusion with SHA256. Something in me thought CIDRAM used CRC. LOL Then again, and now that I think of it, I don't think CRC produces such a long character string. LOL I should have known...

 :four_leaf_clover:  :rainbow:
